File: AspNetMvcUnleashed/0672329980_source/Chapter15/CS/MvcApplication1/MvcApplication1/Controllers/MovieController.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class MovieController : Controller
    {
        MoviesDBEntities _entities = new MoviesDBEntities(); 

        public ActionResult Index()
        {
            return View();
        }

        public ActionResult Refresh()
        {
            return Json( GetThreeMovies() );
        }


        private IEnumerable GetThreeMovies()
        {
            var rnd = new Random();
            var allMovies = _entities.MovieSet.ToList();
            var selectedMovies = new List<Movie>();

            for (int i = 0; i < 3; i++)
            {
                var selected = allMovies[rnd.Next(allMovies.Count)];
                allMovies.Remove(selected);
                selectedMovies.Add(selected);
            }

            var results = from m in selectedMovies
                          select new {Title=m.Title, Director=m.Director};

            return results;
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter15/VB/MvcApplication1/MvcApplication1/Controllers/MovieController.vb
Public Class MovieController
    Inherits Controller

    Private _entities As New MoviesDBEntities()

    Public Function Index() As ActionResult
        Return View()
    End Function

    Public Function Refresh() As ActionResult
        Return Json(GetThreeMovies())
    End Function


    Private Function GetThreeMovies() As IEnumerable
        Dim rnd = New Random()
        Dim allMovies = _entities.MovieSet.ToList()
        Dim selectedMovies = New List(Of Movie)()

        For i As Integer = 0 To 2
            Dim selected = allMovies(rnd.Next(allMovies.Count))
            allMovies.Remove(selected)
            selectedMovies.Add(selected)
        Next i

        Dim results = From m In selectedMovies _
                      Select New With {Key .Title = m.Title, Key .Director = m.Director}

        Return results
    End Function

End Class
